portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,14 +36,17 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get all comments by timestamp to keep order of submission
-    // Sorted newest first so new conversation is not buried under older posts
+    int commentAmount = getCommentAmount(request);
+
+    // Get 'commentAmount' of comments sorted by timestamp
+    // Newest first so new conversation is not buried under older posts
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    PreparedQuery pq = datastore.prepare(query);
+    List<Entity> results = pq.asList(FetchOptions.Builder.withLimit(commentAmount));
 
     List<String> comments = new ArrayList<>();
```

Comment:
Optional stretch goal: You can do this entire segment (extracting the message text from each comment row) in 1 or 2 lines using Java streams. https://winterbe.com/posts/2014/07/31/java8-stream-tutorial-examples/

This is a more advanced topic so feel free to leave it as-is for now. Streams are widely used inside Google (including in Javascript) so they're a helpful thing to learn at some point this summer

Reply:
This feedback is great! Thank you for the link I have replaced it with a stream.